rclcpp/src/rclcpp/time_source.cpp

```diff
@@ -285,6 +309,10 @@ class TimeSource::NodeState final
     // can't possibly call any of the callbacks as we are cleaning up.
     destroy_clock_sub();
     clocks_state_.disable_ros_time();
+    if (on_set_parameters_callback_ && node_parameters_) {
```

Comment:
it seems like other parts of this class are not checking for whether `node_parameters_` is a valid pointer or not

Reply:
True. I was following how it's done in 9723576821304add8993c7e9a3057fa284c1146f, but I agree that we should add the check in `attachNode()`.

Reply:
Removed the null check for node_parameters_ since other parts of the code don't usually perform such a check.